### DISCUSSION TITLE
What's the reason for the Question Generator subsystem?

### DISCUSSION THREAD

**Discussion by greyidol** [Retrospring/retrospring#1110] [General]
I was surprised to find that rather than a flat list of questions, the questions are stored in a yaml file, arranged in a tree according to the words of the question. Why this instead of a plain text file?

**Comment by pixeldesu** [comment]
Instead of defining a bunch of questions like so:

```yaml
- Do you like X
- Do you like Y
- Do you like Z
```

it prevents repetition if we only need to define it like so

```yaml
- Do you like:
  - X
  - Y
  - Z
```

and every branch becomes a question, allowing sets of questions with similar sections to be only defined once.

You could technically just go ahead and define questions one-by-one in the YAML file as well, if you wanted to.

**Reply by greyidol** [reply]
It seems to me like it is a reduction in
a) readability
b) searchability
and c) maintenance.

For a) it is hard to scan a line and see what the question is.
For b) it is hard to search for a phrase, because the phrase is broken up across multiple lines
For c) it is hard to find the place where a new question should go, in part because of (b) and in part because the file can't just be sorted.

Also the sorting problem can be solved by doing something like this https://github.com/ifd3f/year-of-bot/blob/fe984863d36e67f69d246d93397a15b33753c024/sort-data.sh
